runtime/src/epoch_rewards_hasher.rs

```diff
@@ -0,0 +1,25 @@
+use {
+    siphasher::sip::SipHasher13,
+    solana_sdk::{hash::Hash, pubkey::Pubkey},
+    std::hash::Hasher,
+};
+
+/// Use SipHasher13 keyed on the `seed` for calculating epoch reward partition
+pub(crate) fn create_epoch_reward_hasher(hash: &Hash) -> SipHasher13 {
+    let mut hasher = SipHasher13::new();
+    hasher.write(hash.as_ref());
+    hasher
+}
+
+/// Return partition index (0..partitions) by hashing `address` with the `hasher`
+pub(crate) fn address_to_partition<H: Hasher + Copy>(
+    mut hasher: H,
+    partitions: usize,
+    address: &Pubkey,
+) -> usize {
+    hasher.write(address.as_ref());
+    let hash64 = hasher.finish();
+    ((partitions as u128)
+        .saturating_mul(hash64 as u128)
+        .saturating_div((u64::MAX as u128).saturating_add(1))) as usize
+}
```

Comment:
```suggestion
#[derive(Debug, Clone)]
pub(crate) struct EpochRewardHasher{
    hasher: SipHasher13,
    partitions: usize,
}

impl EpochRewardHasher {
    /// Use SipHasher13 keyed on the `seed` for calculating epoch reward partition
    pub fn new(partitions: usize, hash: &Hash) -> Self {
        let mut hasher = SipHasher13::new();
        hasher.write(hash.as_ref());
        Self { hasher, partitions }
    }

    /// Return partition index (0..partitions) by hashing `address` with the `hasher`
    pub fn hash_address_to_partition(self, address: &Pubkey) -> usize {
        let Self { mut hasher, partitions } = self;
        hasher.write(address.as_ref());
        let hash64 = hasher.finish();
        ((partitions as u128)
            .saturating_mul(u128::from(hash64))
            .saturating_div(u128::from(u64::MAX).saturating_add(1))) as usize
    }
}
```

better wrapped in a type? i can't tell. interface is a little tighter, at least :thinking:

Reply:
yeah. done